docs/java/how-to-get-the-result-of-an-activity-execution-in-java.md

```diff
@@ -0,0 +1,53 @@
+---
+id: how-to-get-the-result-of-an-activity-execution-in-java
+title: How to get the result of an Activity Execution
+sidebar_label: Activity Execution Result
+description: To get the results of an asynchronously invoked Activity method, use the `Promise` `get` method to block until the Activity method result is available.
+tags:
+  - java
+  - developer-guide
+---
+
+To get the results of an asynchronously invoked Activity method, use the `Promise` `get` method to block until the Activity method result is available.
+
+Sometimes an Activity lifecycle goes beyond a synchronous method invocation.
+For example, a request can be put in a queue and later a reply comes and is picked up by a different Worker process.
+The whole request-reply interaction can be modeled as a single Activity.
+
+To indicate that an Activity should not be completed upon its method return, call `ActivityExecutionContext.doNotCompleteOnReturn()` from the original Activity thread.
+
+Then later, when replies come, complete the Activity using the `ActivityCompletionClient`.
+To correlate Activity invocation with completion use either a `TaskToken` or Workflow and Activity IDs.
+
+Following is an example of using `ActivityExecutionContext.doNotCompleteOnReturn()`:
+
+```java
+public class FileProcessingActivitiesImpl implements FileProcessingActivities {
```

Comment:
The file processing example is far more complex as an example than what we need here.

Reply:
true; keeping it for now but will update it soon.